contrib/fsl_integration/sct_plugin.py

```diff
@@ -216,44 +285,530 @@ def __init__(self, parent):
+    <b>Specific citation</b>:<br>
+    Xu et al.
+    <i>Evaluation of slice accelerations using multiband echo planar imaging at 3 T.
+    (2013)</i>. Neuroimage.
+
+    """
+
+    def __init__(self, parent):
+        super(TabPanelMOCO, self).__init__(parent=parent,
+                                            id_=wx.ID_ANY)
+        button_gm = wx.Button(self, id=wx.ID_ANY,
+                              label="dMRI Moco")
+        button_gm.Bind(wx.EVT_BUTTON, self.onButtonMOCO)
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(button_gm, 0, wx.ALL, 5)
+        self.sizer_h.Add(sizer)
+        self.SetSizerAndFit(self.sizer_h)
+
+    def onButtonMOCO(self, event):
+        selected_overlay = displayCtx.getSelectedOverlay()
+        filename_path = selected_overlay.dataSource
+
+        path_name = os.path.dirname(filename_path)
+        base_name = os.path.basename(filename_path)
+
+        fname, fext = base_name.split(os.extsep, 1)
+
+        mean_name = "{}_mean.{}".format(fname,fext)
+        meanname_path = os.path.join(path_name, mean_name)
+
+        seg_name = "{}_mean_seg.{}".format(fname, fext)
+        segname_path = os.path.join(path_name, seg_name)
+
+        mask_name = "mask_{}_mean.{}".format(fname,fext)
+        maskname_path = os.path.join(path_name, mask_name)
+
+        crop_name = "{}_crop.{}".format(fname,fext)
+        cropname_path = os.path.join(path_name, crop_name)
+
+        bvec_name = "{}_bvecs.txt".format(fname)
+        bvecname_path = os.path.join(path_name, bvec_name)
+
+        out_name = "{}_crop_moco.{}".format(fname, fext)
+        outname_path = os.path.join(path_name, out_name)
+
+        cmd_line1 = "sct_maths -i {} -mean t -o {}".format(filename_path, meanname_path)
+        print('Command:', cmd_line1)
+        self.call_sct_command(cmd_line1)
+
+        image = Image(meanname_path)
+        overlayList.append(image)
+        del overlayList[0]
+
+        opts = displayCtx.getOpts(image)
+        opts.cmap = 'gray'
+
+        cmd_line2 = "sct_deepseg_sc -i {} -c dwi".format(meanname_path)
+        print('Command:', cmd_line2)
+        self.call_sct_command(cmd_line2)
+
+        image = Image(segname_path)
+        overlayList.append(image)
+
+        opts = displayCtx.getOpts(image)
+        opts.cmap = 'red'
+
+        cmd_line3 = "sct_create_mask -i {} -p centerline,{} -size 35mm".format(meanname_path, segname_path)
+        print('Command:', cmd_line3)
+        self.call_sct_command(cmd_line3)
+
+        del overlayList[1]
+
+        image = Image(maskname_path)
+        overlayList.append(image)
+
+        opts = displayCtx.getOpts(image)
+        opts.cmap = 'red'
+
+        cmd_line4 = "sct_crop_image -i {} -m {} -o {}".format(filename_path, maskname_path, cropname_path)
+        print('Command:', cmd_line4)
+        self.call_sct_command(cmd_line4)
+
+        cmd_line5 = "sct_dmri_moco -i {} -bvec {}".format(cropname_path, bvecname_path)
```

Comment:
indicate how should the bvals/bvecs be named, and also make the possibility to add a mask